Restore TestMtlsHealthCheck in postsubmit, prow.

incfly opened the pull request:
https://github.com/istio/istio/issues/12754

- I changed this to only run in post submit, using prow. This runs 4 times and all succeed. https://k8s-gubernator.appspot.com/builds/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/
- Presubmit test, using circleci and minikube, continuously timeout in healthcheck app deployment.
Given that, I changed this to only run in postsubmit for now.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/12969#" title="Author self-approved">incfly</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ozevren**

If they are not already assigned, you can assign the PR to them by writing `/assign @ozevren` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[tests/integration/OWNERS](https://github.com/istio/istio/blob/master/tests/integration/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["ozevren"]} -->

istio-testing commented:
@incfly: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 355f4a8734244fa4f87059470d4bac54a7621425 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10422) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

incfly commented:
@ozevren here's what i saw from the presubmit tests:

- prow/integ-k8s pass, including this one, `TestMtlsHeathCheck`.
- All tests from `ci/circleci: test-integration-kubernetes` are skipping.

```
=== RUN   TestIngessToPrometheus_IngressMetric
--- SKIP: TestIngessToPrometheus_IngressMetric (0.00s)
	testcontext.go:54: Skipping: label mismatch: labels=map[], filter=+presubmit
```

Is that intentional? Do we tweak something for circleci to skip all the tests? Should I then just focus on stabilizing prow tests?

@utka

ozevren commented:
Hey, sorry for not publicisizing this.

I made a change on Friday, so that tests get selected into different suites based on labels. I've setup the CircleCI tests, so that only the tests marked with labels.Presubmit get selected (which includes tests I know for sure are healthy). The Prow tests run everything. 

My intention is to make the CircleCI tests as a "presubmit" gate, and use the Prow one to use as a baking ground / for fixing flakes before promoting to presubmit. , I am hoping to change the test names soon to better reflect what is going on.

Sorry for the confusion. I'll add some documentation soon and update the test names accordingly to reflect this.

incfly commented:
Thanks for explaining. So sounds like we prefer to use circleci for stable test, "required".

istio-testing commented:
@incfly: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | c66ff154f5a87e6a7f0b7af63030667180270126 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10456) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

incfly commented:
/test istio-integ-k8s-tests

incfly commented:
/test istio-integ-k8s-tests

incfly commented:
- I changed this to only run in post submit, using prow. This runs 4 times and all succeed.
- Presubmit test, using circleci and minikube, continuously timeout in healthcheck app deployment.

Given that, I changed this to only run in postsubmit for now.

istio-testing commented:
@incfly: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 322502f4a6821d868008ac765f9f44921e43bf36 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10474) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

incfly commented:
/hold cancel

PTAL, @ozevren thanks

ozevren commented on tests/integration/security/healthcheck/mtls_healthcheck_test.go:
```diff
@@ -34,8 +35,13 @@ var (
+	// For more details, see https://github.com/istio/istio/issues/12754.
 	framework.NewSuite("mtls_healthcheck", m).
 		RequireEnvironment(environment.Kube).
+		Label(label.Postsubmit).
```
The Postsubmit label hasn't been wired yet. Just removing Skip below should be enough to get the test running in the queues (non-required presubmit & postsubmit) again.

incfly replied:
Done.

incfly replied:
I believe this will make test run in prow/circleci-integ-k8s since circleci currently it's empty...

ozevren commented:
/lgtm

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/12969#" title="Author self-approved">incfly</a>*, *<a href="https://github.com/istio/istio/pull/12969#issuecomment-479604748" title="LGTM">ozevren</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[tests/integration/OWNERS](https://github.com/istio/istio/blob/master/tests/integration/OWNERS)~~ [ozevren]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 322502f4a6821d868008ac765f9f44921e43bf36 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10474) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-unit-tests/23246) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-unit-tests/23246) | `/test istio-unit-tests`
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10513) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

incfly commented:
/test istio-unit-tests
--

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-unit-tests/23246) | `/test istio-unit-tests`
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10513) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6163) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6163) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10522) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10522) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6174) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6174) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10527) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10527) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6178) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10527) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6178) | `/test istio-integ-k8s-tests`
prow/e2e_pilotv2_auth_sds.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio_auth_sds_e2e/4147) | `/test istio_auth_sds_e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

incfly commented:
/test e2e_pilotv2_auth_sds

incfly commented:
/test istio_auth_sds_e2e

incfly commented:
/retest

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6178) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10544) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@incfly: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-pilot-multicluster-e2e/10544) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | f9d05fda4f80dbc3b4e814ab7790546bdd24f7f2 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12969/istio-integ-k8s-tests/6199) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
